src/OrleansProviders/Streams/Generator/GeneratorPooledCache.cs

```diff
@@ -58,13 +60,65 @@ public bool Equals(CachedMessage cachedMessage, IStreamIdentity streamIdentity)
             }
         }
 
+        private class ExplicitEvictionStrategy : IEvictionStrategy<CachedMessage>
```

Comment:
I thought about that. but typed param made it harder to make common. Since we need to know what type of CachedMessage is to know its buffer ID